src/main/java/com/google/jenkins/plugins/storage/StdoutUploadStep.java

```diff
@@ -0,0 +1,186 @@
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import jenkins.tasks.SimpleBuildStep;
+import net.sf.json.JSONObject;
+import org.jenkinsci.Symbol;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+import org.kohsuke.stapler.QueryParameter;
+import org.kohsuke.stapler.StaplerRequest;
+
+/**
+ * Build Step wrapper for StdoutUpload to be run in pipelines. Should ideally be run post build
+ * instead of a regular post step.
+ */
+@RequiresDomain(StorageScopeRequirement.class)
+public class StdoutUploadStep extends Builder implements SimpleBuildStep, Serializable {
+
+  @Nonnull private StdoutUpload upload;
+
+  @DataBoundConstructor
+  public StdoutUploadStep(String credentialsId, String bucket, String logName) {
+    this(credentialsId, bucket, null, logName);
+  }
+
+  /** Construct the build log upload step. */
+  public StdoutUploadStep(
+      String credentialsId, String bucket, @Nullable UploadModule module, String logName) {
+    this.credentialsId = credentialsId;
+    upload = new StdoutUpload(bucket, module, logName, null);
+
+    // Build steps will not be executed following a failed build.
+    // Pipeline steps performed sequentially will not be executed
+    //   following a failed step
+    // If we ever get to execute this on a failed build, that must
+    // have been done intentionally, e.g., using "post" with appropriate
+    // flags. This should be allowed.
+    upload.setForFailedJobs(true);
+  }
+
+  /** Whether to surface the file being uploaded to anyone with the link. */
+  @DataBoundSetter
+  public void setSharedPublicly(boolean sharedPublicly) {
+    upload.setSharedPublicly(sharedPublicly);
+  }
+
+  public boolean isSharedPublicly() {
+    return upload.isSharedPublicly();
+  }
+
+  /**
+   * Whether to indicate in metadata that the file should be viewable inline in web browsers, rather
+   * than requiring it to be downloaded first.
+   */
+  @DataBoundSetter
+  public void setShowInline(boolean showInline) {
+    upload.setShowInline(showInline);
+  }
+
+  public boolean isShowInline() {
+    return upload.isShowInline();
+  }
+
+  /**
+   * The path prefix that will be stripped from uploaded files. May be null if no path prefix needs
+   * to be stripped.
+   *
+   * <p>Filenames that do not start with this prefix will not be modified. Trailing slash is
+   * automatically added if it is missing.
+   */
+  @DataBoundSetter
+  public void setPathPrefix(@Nullable String pathPrefix) {
+    upload.setPathPrefix(pathPrefix);
+  }
+
+  @Nullable
+  public String getPathPrefix() {
+    return upload.getPathPrefix();
+  }
+
+  public String getLogName() {
+    return upload.getLogName();
+  }
+
+  public String getBucket() {
+    return upload.getBucket();
+  }
+
+  /** The unique ID for the credentials we are using to authenticate with GCS. */
+  public String getCredentialsId() {
+    return credentialsId;
+  }
+
+  private final String credentialsId;
```

Comment:
Instance variable declarations should be grouped together at the beginning of the class.

Reply:
Done